kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/installation/kroxylicious/Kroxylicious.java

```diff
@@ -51,10 +54,22 @@ private void createTopicEncryptionFilterConfigMap(String clusterName, String top
 
     private void deployPortPerBrokerPlain(int replicas) {
         LOGGER.info("Deploy Kroxylicious in {} namespace", deploymentNamespace);
+        createSecrets();
         resourceManager.createResourceWithWait(KroxyliciousDeploymentTemplates.defaultKroxyDeployment(deploymentNamespace, containerImage, replicas).build());
         resourceManager.createResourceWithoutWait(KroxyliciousServiceTemplates.defaultKroxyService(deploymentNamespace).build());
     }
 
+    /**
+     * Needed for downstream Kroxylicious images
+     */
+    private void createSecrets() {
+        String configFolder = Environment.CONTAINER_CONFIG_PATH;
+        SecretBuilder secretBuilder = KroxyliciousSecretTemplates.createRegistryCredentialsSecret(configFolder, deploymentNamespace);
```

Comment:
Isn't the same pattern needed for Strimzi in the case where it comes from a private registry too?   That makes wonder if you should centralise the responsibility.

Reply:
Yeap, my fault. I have added some needed changes into the current jenkins pipeline and it worked. I will close this PR once the MR on strimzi-ci is merged

Reply:
> I will close this PR once the MR on strimzi-ci is merged

I don't really understand this comment.  Are you saying this PR is redundant?  I guess no

Reply:
If we use the same method than Strimzi, the config json is created before creating the kubernetes cluster, so everything is copied inside directly. I did that with the secrets, because we don't have that step locally where to copy everything directly on kubernetes and we would need to create the secret.

So if we want to have something that allows us to run the tests locally, we should keep this PR, if not we can avoid introducing more unnecessary code.

Reply:
https://master-jenkins-csb-amq-streams.apps.ocp-c1.prod.psi.redhat.com/job/kroxylicious/job/kroxylicious-system-tests/1/ here is an execution of the test cases on current main, using only the Strimzi-ci approach. No secrets needed

Reply:
So, would for minikube users, would this have the same effect?

https://minikube.sigs.k8s.io/docs/handbook/registry/#using-a-private-registry

If yes, or it is otherwise easy, then I don't think we should merge this PR.